### ISSUE TITLE
no detach in critic network, Is it a bug?

### ISSUE THREAD

**Issue by KK666-AI** [shariqiqbal2810/maddpg-pytorch#11]
Dear Authors,

Thanks a lot for your released code. It's very helpful.

I notice that when updating parameters in current_agent's critic, you didn't detach other agents' action. 

`all_pol_acs.append(onehot_from_logits(pi(ob)))`

`all_pol_acs.append(pi(ob))`

Does it should be 

`all_pol_acs.append(onehot_from_logits(pi(ob).detach()))`

`all_pol_acs.append(pi(ob).detach())`

According to my acknowledgement, you update each agent's critic separately. If so, other gent's critic should be fixed. Right?

**Comment by shariqiqbal2810** [comment]
In the case of discrete policies, using `onehot_from_logits` detaches the graph. In the case of continuous policies, detaching would be correct; however, it shouldn't make a difference since the gradients are [zeroed out](https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L134) before each agent's policy update. Adding the `.detach()` may give a performance boost though.
